substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/xml/JavaxXmlClassAndResourcesLoaderFeature.java

```diff
@@ -36,13 +36,24 @@
 
 import com.oracle.svm.core.feature.InternalFeature;
 import com.oracle.svm.core.jdk.JNIRegistrationUtil;
+import com.oracle.svm.hosted.classinitialization.ClassInitializationSupport;
 import com.oracle.svm.core.feature.AutomaticallyRegisteredFeature;
 
 @AutomaticallyRegisteredFeature
 public class JavaxXmlClassAndResourcesLoaderFeature extends JNIRegistrationUtil implements InternalFeature {
 
     @Override
     public void beforeAnalysis(BeforeAnalysisAccess access) {
+        /*
+         * Ensure that class loading of xml related classes work.
+         * com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl implicitly depends
+         * on jdk.xml.internal.JdkXmlUtils. So initialization order needs to happen in the reverse
+         * order.
+         */
+        Class<?> jdkUtil = access.findClassByName("jdk.xml.internal.JdkXmlUtils");
+        if (jdkUtil != null) {
+            ClassInitializationSupport.singleton().forceInitializeHosted(jdkUtil, "break circular class initialization in XML code", false);
```

Comment:
I created #8256 because then the registration of the individual class is at the same place that registers the whole package for initialization at build time, i.e., when we remove the package we will remember to also remove the class.

Reply:
Sounds good. Thank you.

Reply:
https://github.com/oracle/graal/pull/8256 fixed this.